### ISSUE TITLE
DECIMAL comparison is not handling rescaling

### ISSUE THREAD

**Issue by Yuhta** [facebookincubator/velox#3133]
In #2543 we have simple function implementation comparing unscaled decimals directly, which is not working if the scales are not matching.  The functions should consider rescaling.

@majetideepak @karteekmurthys

**Comment by mbasmanova** [comment]
CC: @aditi-pandit

**Comment by karteekmurthys** [comment]
@Yuhta @mbasmanova @majetideepak 
This as per the design. We expect the user to user cast decimal values to same scale before calling comparison operations: 

https://github.com/facebookincubator/velox/pull/2278

"In this PR, we introduce a way to declare SCALAR functions for one or more decimal type inputs with same precision and scale. To achieve this, we inject default signatures SHORT_DECIMAL(a_precision,a_scale),
LONG_DECIMAL(a_precision,a_scale) for all decimal types in the TypeAnalysis phase."

**Comment by Yuhta** [comment]
@karteekmurthys Thanks for the explanation.  Do we have any plan to support directly comparison between DECIMALs without casting?

**Comment by mbasmanova** [comment]
CC: @laithsakka

**Comment by karteekmurthys** [comment]
@Yuhta We did consider designing Decimal Comparison Vector functions similar to decimal arithmetic, but we decided to leverage the Generic Framework over vector functions. Makes it easier to manage different vector types.

We also observed that the plan from Presto does cast the parameters to same type to comparison functions:
```
presto:kart_test> select * from tA where c0 between cast(0.01 as decimal(3,2)) and cast(0.8 as decimal(2,1));
   c0    
---------
 0.12345
 0.23456
 0.45678
(3 rows)
```
The predicate “BETWEEN” plan node looks as follows, The column C0 is of type DECIMAL(5,5) and the constants are cast to column’s type.



<img width="818" alt="Screen Shot 2022-11-07 at 2 32 17 PM" src="https://user-images.githubusercontent.com/6971561/200429640-87f22d3d-48ee-4a83-bd26-e4d3406ce333.png">
